sdk/src/types/block/output/delegation.rs

```diff
@@ -0,0 +1,599 @@
+    pub fn with_unlock_conditions(
+        mut self,
+        unlock_conditions: impl IntoIterator<Item = impl Into<UnlockCondition>>,
+    ) -> Self {
+        self.unlock_conditions = unlock_conditions.into_iter().map(Into::into).collect();
+        self
+    }
+
+    /// Replaces an [`UnlockCondition`] of the builder with a new one, or adds it.
+    pub fn replace_unlock_condition(mut self, unlock_condition: impl Into<UnlockCondition>) -> Self {
+        self.unlock_conditions.replace(unlock_condition.into());
+        self
+    }
+
+    /// Clears all [`UnlockConditions`]s from the builder.
+    pub fn clear_unlock_conditions(mut self) -> Self {
+        self.unlock_conditions.clear();
+        self
+    }
+
+    /// Adds an immutable [`Feature`] to the builder, if one does not already exist of that type.
+    pub fn add_immutable_feature(mut self, immutable_feature: impl Into<Feature>) -> Self {
+        self.immutable_features.insert(immutable_feature.into());
+        self
+    }
+
+    /// Sets the immutable [`Feature`]s in the builder, overwriting any existing values.
+    pub fn with_immutable_features(mut self, immutable_features: impl IntoIterator<Item = impl Into<Feature>>) -> Self {
+        self.immutable_features = immutable_features.into_iter().map(Into::into).collect();
+        self
+    }
+
+    /// Replaces an immutable [`Feature`] of the builder with a new one, or adds it.
+    pub fn replace_immutable_feature(mut self, immutable_feature: impl Into<Feature>) -> Self {
+        self.immutable_features.replace(immutable_feature.into());
+        self
+    }
+
+    /// Clears all immutable [`Feature`]s from the builder.
+    pub fn clear_immutable_features(mut self) -> Self {
+        self.immutable_features.clear();
+        self
+    }
+
+    /// Finishes the builder into a [`DelegationOutput`] without amount verification.
+    pub fn finish_unverified(self) -> Result<DelegationOutput, Error> {
+        let unlock_conditions = UnlockConditions::from_set(self.unlock_conditions)?;
+
+        verify_unlock_conditions::<true>(&unlock_conditions)?;
+
+        let immutable_features = Features::from_set(self.immutable_features)?;
+
+        verify_allowed_features(&immutable_features, DelegationOutput::ALLOWED_IMMUTABLE_FEATURES)?;
+
+        let mut output = DelegationOutput {
+            amount: 1u64,
+            delegated_amount: self.delegated_amount,
+            delegation_id: self.delegation_id,
+            validator_id: self.validator_id,
+            start_epoch: self.start_epoch,
+            end_epoch: self.end_epoch,
+            unlock_conditions,
+            immutable_features,
+        };
+
+        output.amount = match self.amount {
+            OutputBuilderAmount::Amount(amount) => amount,
+            OutputBuilderAmount::MinimumStorageDeposit(rent_structure) => {
+                Output::Delegation(output.clone()).rent_cost(&rent_structure)
```

Comment:
The rent_cost() needs to be updated when it's defined how exactly, but maybe put a TODO to it?

Reply:
Sure, for now I'm only adding the layouts and not doing anything related to verification

Reply:
Fair enough, just think this is something that's rather easy to forget, but no TODO in code needed, added an entry in https://github.com/iotaledger/iota-sdk/issues/647

Reply:
Thanks